File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/jobinfo/ui/JobInfoPresenterModule.java
package com.fintechplatform.ui.profile.jobinfo.ui;

import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.api.profile.api.ProfileAPI;
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class JobInfoPresenterModule {

    private JobInfoContract.View view;
    private DataAccount configuration;

    public JobInfoPresenterModule(JobInfoContract.View view, DataAccount configuration) {
        this.view = view;
        this.configuration = configuration;
    }

    @Provides
    @Singleton
    JobInfoContract.Presenter providesJobInfoPresenter(ProfileAPI api, UsersPersistanceDB usersPersistanceDB) {
        return new JobInfoPresenter(view, api, configuration, usersPersistanceDB);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/jobinfo/di/JobInfoPresenterModule.kt
package com.fintechplatform.ui.profile.jobinfo.di

import com.fintechplatform.api.profile.api.ProfileAPI
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB
import com.fintechplatform.ui.profile.jobinfo.JobInfoContract
import com.fintechplatform.ui.profile.jobinfo.JobInfoPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class JobInfoPresenterModule(private val view: JobInfoContract.View, private val configuration: DataAccount) {

    @Provides
    @Singleton
    internal fun providesJobInfoPresenter(api: ProfileAPI, usersPersistanceDB: UsersPersistanceDB): JobInfoContract.Presenter {
        return JobInfoPresenter(view, api, configuration, usersPersistanceDB)
    }
}